### DISCUSSION TITLE
nuxt 3 + keycloak authorization

### DISCUSSION THREAD

**Discussion by Kingside88** [nuxt/nuxt#18968] [Questions] [answered]
Hi, i am in the situation stuggeling with keycloak.
keycloak is an identity server for handling authorization. 
My first attempt was to use sidebase nuxt-auth alongside keycloke provider.
Yes it works, but is has some bugs like SignOut Session clearing and the JWT token is not valid.
You can check out my ready to go project here: 
https://github.com/Kingside88/nuxt3-primevue-starter-auth

My second try is to implement clean vue3 with keycloakJs. This works very well without bugs.
You can see here the main branch:
https://github.com/Kingside88/vue3-auth-keycloakJs

Now I want to implement nuxt3 with keycloakJs with plugins. But get error 500 - document not defined.
As I know this issue has something to to with client/server compliance.
https://github.com/Kingside88/nuxt3-primevue-starter-auth/tree/keycloakJs

I hope someone can give some help.

**Comment by Kingside88** [comment, marked as answer]
Ok guys, I've got an update for you. Nuxt 3 is running with official keycloakJs Javascript package.
You can get a ready to go demo here: 
https://github.com/Kingside88/nuxt3-primevue-starter-auth/blob/keycloakJs/README.md
The sidebase/nuxt-auth or better say next-auth keycloak provider has a bug providing the JWT token. It misses some properties. Now we can compare both solutions.

**Reply by Yogiii** [reply]
Hello, 
Im trying to add keycloak on my nuxt 3 + vuetify app but facing the same problem as you : document is not define.

I clone your demo project branch keycloakJs, but cannot make it run, first i had  
> Failed to resolve import "keycloak-js" from "app.vue"
Then after installing via pnpm keycloakJs i had 
>  Uncaught (in promise) Error: PrimeVue is not installed!
But same error after pnpm primevue

Could you help me ?

Node version : v18.13.0
pnpm version : 7.30.0

**Reply by Kingside88** [reply]
Please did some changes. Please try again the keyclakJs Branch.
And please create an issue there so I can see it in the future.
